be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -830,7 +830,8 @@ Status BaseBetaRowsetWriter::_build_rowset_meta(RowsetMeta* rowset_meta, bool ch
 
     rowset_meta->set_num_segments(segment_num);
     rowset_meta->set_num_rows(num_rows_written + _num_rows_written);
-    rowset_meta->set_total_disk_size(total_data_size + _total_data_size);
+    rowset_meta->set_total_disk_size(total_data_size + _total_data_size + total_index_size +
```

Comment:
If you want to use the formula ·total size = column data size + index size·, you need to modify the data size and index size statistics in the segment creator, vertical segment writer, and vertical beta rowset writer.

Reply:
Currently, `data size = segment file size + inverted index file size`, `total size = data size`